internal/compiler/passes/lower_component_container.rs

```diff
@@ -3,24 +3,29 @@
 
 use crate::diagnostics::BuildDiagnostics;
 use crate::langtype::ElementType;
-use crate::object_tree::*;
-use crate::typeregister::TypeRegister;
+use crate::typeloader::TypeLoader;
+use crate::{expression_tree, object_tree::*};
+use std::cell::RefCell;
 use std::rc::Rc;
 
 pub fn lower_component_container(
-    component: &Rc<Component>,
-    type_register: &TypeRegister,
+    doc: &Document,
+    type_loader: &mut TypeLoader,
     diag: &mut BuildDiagnostics,
 ) {
-    let empty_type = type_register.empty_type();
+    let empty_type = type_loader.global_type_registry.borrow().empty_type();
 
-    recurse_elem_including_sub_components_no_borrow(component, &None, &mut |elem, _| {
-        if matches!(&elem.borrow().builtin_type(), Some(b) if b.name == "ComponentContainer") {
-            diagnose_component_container(elem, diag);
-            process_component_container(elem, &empty_type);
-        }
-        Some(elem.clone())
-    })
+    let mut new_components = vec![];
```

Comment:
This could be just a counter

```suggestion
    let mut counter = 0;
```

Reply:
The point is that there is no need to have that to be a vector.
The comment make it sound like this is solved, but it is not.

Reply:
I agree. It is *really* gone now :-)